### ISSUE TITLE
Multiple Custom Validation Block

### ISSUE THREAD

**Issue by khpatel4991** [dry-rb/dry-validation#299]
I have a schema in which i need to check for a particular input for two different errors. 

```
p = Dry::Validation.Form do 
  configure do 
    config.type_specs = true
  end
  required(:name).filled(:str?)
  required(:p_p).each { schema pp }

  validate(rule1: :p_p) do |p_p|
    validate_rule1_logic(p_p)
  end

  validate(rule2: :p_p) do |p_p|
    validate_rule2_logic(p_p)
  end
end
```

Now when I have an input which breaks rule1, it throws the error as expected. If I have input which breaks rule2 and passes rule 1, it **wont throw any error**. Now If I change the order of the rules like as follows, the behavior is followed for rule 2.: 

```
  validate(rule2: :p_p) do |p_p|
    validate_rule2_logic(p_p)
  end

  validate(rule1: :p_p) do |p_p|
    validate_rule1_logic(p_p)
  end

```
Am I doing something wrong, am I allowed to have only 1  validate rule per input-key? Or is there any other way I can fix this issue?

**Comment by solnic** [comment]
Looks like a bug. Thanks for reporting this. I'll look into that.

**Comment by nicolas-besnard** [comment]
Just had the [same issue](https://github.com/dry-rb/dry-validation/issues/312).

Any news on this ?

**Comment by solnic** [comment]
I'll fix it in the next version once I find time

**Comment by nicolas-besnard** [comment]
Do you have any idea where this can came from ? I might have some free time to look at this

**Comment by solnic** [comment]
@nicolas-besnard sure, help is always welcome! :)

**Comment by kimquy** [comment]
We probably can close this one also cc @solnic ?

**Comment by solnic** [comment]
Yes! Unfortunately I released 0.10.6 before I noticed your PR. I'll push 0.10.7 with your fix on Friday.
